sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/AmqpEventHubConsumerTests.cs

```diff
@@ -0,0 +1,347 @@
+        /// </summary>
+        ///
+        [Test]
+        public void ConstructorRequiresTheOptions()
+        {
+            Assert.That(() => new AmqpEventHubConsumer("theMostAwesomeHubEvar", "$DEFAULT", "0", EventPosition.FromOffset(1), null, Mock.Of<AmqpConnectionScope>(), Mock.Of<AmqpMessageConverter>(), Mock.Of<EventHubRetryPolicy>(), new LastEnqueuedEventProperties("hub", "0")), Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the constructor.
+        /// </summary>
+        ///
+        [Test]
+        public void ConstructorRequiresTheConnectionScope()
+        {
+            Assert.That(() => new AmqpEventHubConsumer("theMostAwesomeHubEvar", "$DEFAULT", "0", EventPosition.FromSequenceNumber(123), new EventHubConsumerOptions(), null, Mock.Of<AmqpMessageConverter>(), Mock.Of<EventHubRetryPolicy>(), null), Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the constructor.
+        /// </summary>
+        ///
+        [Test]
+        public void ConstructorRequiresTheRetryPolicy()
+        {
+            Assert.That(() => new AmqpEventHubConsumer("theMostAwesomeHubEvar", "$DEFAULT", "0", EventPosition.Latest, new EventHubConsumerOptions(), Mock.Of<AmqpConnectionScope>(), Mock.Of<AmqpMessageConverter>(), null, new LastEnqueuedEventProperties("hub", "0")), Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpEventHubConsumer.CloseAsync"/>
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public async Task CloseMarksTheConsumerAsClosed()
+        {
+            var consumer = new AmqpEventHubConsumer("aHub", "$DEFAULT", "0", EventPosition.Earliest, new EventHubConsumerOptions(), Mock.Of<AmqpConnectionScope>(), Mock.Of<AmqpMessageConverter>(), Mock.Of<EventHubRetryPolicy>(), null);
+            Assert.That(consumer.Closed, Is.False, "The consumer should not be closed on creation");
+
+            await consumer.CloseAsync(CancellationToken.None);
+            Assert.That(consumer.Closed, Is.True, "The consumer should be marked as closed after closing");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpEventHubConsumer.CloseAsync"/>
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void CloseRespectsTheCancellationToken()
+        {
+            var consumer = new AmqpEventHubConsumer("aHub", "$DEFAULT", "0", EventPosition.Earliest, new EventHubConsumerOptions(), Mock.Of<AmqpConnectionScope>(), Mock.Of<AmqpMessageConverter>(), Mock.Of<EventHubRetryPolicy>(), null);
+            using var cancellationSource = new CancellationTokenSource();
+
+            cancellationSource.Cancel();
+            Assert.That(async () => await consumer.CloseAsync(cancellationSource.Token), Throws.InstanceOf<TaskCanceledException>());
```

Comment:
It might be a good idea to assert that `consumer.Closed` is still false.

Reply:
good call.   Done.